at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -30,6 +33,7 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
   late final AtClient _atClient;
   late final RemoteSecondary _remoteSecondary;
   late final NotificationServiceImpl _statsNotificationListener;
+  final List<SyncProgressListener> _syncProgressListeners = [];
```

Comment:
Let's make it a set to prevent duplicate listeners being added.

Reply:
done